### DISCUSSION TITLE
Is there a way to mark an object as raw to not unwrap it when returning from `setup()`?

### DISCUSSION THREAD

**Discussion by matthew-dean** [vuejs/rfcs#619] [General]
I have a class that has properties that are Vue refs. In some instances, I pass the ref properties to sub-components / classes.

However, this:
```ts
setup() {
    return new MyViewModel()
  },
```
...will shallowly unwrap the instance's properties so that refs are now their `.value`s. This is, of course, consistent with [the documentation](https://vuejs.org/api/composition-api-setup.html#basic-usage).

However, when researching this, and coming across `markRaw`, I expected this instance to remain untouched after being returned from setup. I did this:
```ts
setup() {
    return markRaw(new MyViewModel())
  },
```

To my surprise, even though this was marked as raw, the instance was _still_ shallowly unwrapped. Is there a way to prevent this?

**Comment by sevs44936** [comment]
I've run into the same problem of `markRaw(ref( any value ))` still getting unwrapped by the proxy. The only solution I've found is to wrap the Ref in an extra object I've marked as raw:
```js
const no_unwrapping_refs = ref({
    raw_thingies: markRaw({ // <-- extra object
        thing: ref("value")
    })
})
```
It looks like an ugly hack, but I haven't found a better solution.
